### DISCUSSION TITLE
Delay data collection during optimization

### DISCUSSION THREAD

**Discussion by Sephiram591** [NanoComp/meep#2109] [Q&A]
When optimizing, I would like to delay when the field monitors begin recording data up till a certain time in the simulation. Is this feature already possible in the monitor instantiation or in the OptimizationProblem class?

**Comment by smartalecH** [comment]
Not currently. Whats your motivation?

**Reply by Sephiram591** [reply]
We are trying to analyze the fields of a ring resonator after the more transient portions of our gaussian source have escaped the simulation, in order to include a cost function to improve purcell enhancement at our target frequency.

**Reply by stevengj** [reply]
Purcell enhancement is just the LDOS, i.e. the work done by the current.  To compute this accurately the simplest way is to directly compute J•E in the frequency domain, which requires the Fourier transform of the full time series.  (In fact, the latest `master` version of the code includes a built-in differentiable LDOS feature.)

See also [this paper](http://doi.org/10.1364/OE.21.030812) and [this paper](https://aip.scitation.org/doi/10.1063/1.5064468) on topology optimization of cavities by LDOS optimization.

**Reply by Andeloth** [reply]
When are the newest changes planned to be released in an official version update (1.24)? We’re running the code on a supercomputer so building from source would be a challenge, but we’d like to use the new LDOS objective in the adjoint solver.

**Reply by Sephiram591** [reply]
Thank you for the swift response! Like Ben said, we look forward to a version update that will allow enable us to use that more easily.
